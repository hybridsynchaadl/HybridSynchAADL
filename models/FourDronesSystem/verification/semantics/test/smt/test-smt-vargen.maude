
set show timing off .

load ../../interpreter-symbolic2.maude

select SMT-VAR-GEN .

red type([true]) .
red type([0/1]) .
red type(*) .


red gen(< "a", 0, 0 >, Boolean) .
red gen(< "a", 0, 0 >, Real) .
red gen(< "a", 0, 0 >, Unit) .


red fresh?(r(0), < "a", 1, 2 >)  .
red fresh?(r(1), < "a", 1, 2 >)  .
red fresh?(b(0), < "a", 1, 2 >)  .
red fresh?(b(1), < "a", 1, 2 >)  .
red fresh?(b(2), < "a", 1, 2 >)  .


