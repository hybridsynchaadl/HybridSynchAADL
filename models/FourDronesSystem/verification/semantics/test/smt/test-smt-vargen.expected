==========================================
reduce in SMT-VAR-GEN : type([(true).Boolean]) .
rewrites: 1
result DataType: Boolean
==========================================
reduce in SMT-VAR-GEN : type([(0/1).Real]) .
rewrites: 1
result DataType: Real
==========================================
reduce in SMT-VAR-GEN : type(*) .
rewrites: 1
result DataType: Unit
==========================================
reduce in SMT-VAR-GEN : gen(< "a",0,0 >, Boolean) .
rewrites: 7
result Pair{Exp,VarGen}: {[b("a#$#0")],< "a",0,1 >}
==========================================
reduce in SMT-VAR-GEN : gen(< "a",0,0 >, Real) .
rewrites: 7
result Pair{Exp,VarGen}: {[r("a#$#0")],< "a",1,0 >}
==========================================
reduce in SMT-VAR-GEN : gen(< "a",0,0 >, Unit) .
rewrites: 1
result Pair{Exp,VarGen}: {*,< "a",0,0 >}
==========================================
reduce in SMT-VAR-GEN : fresh?(r(0), < "a",1,2 >) .
rewrites: 19
result Bool: (false).Bool
==========================================
reduce in SMT-VAR-GEN : fresh?(r(1), < "a",1,2 >) .
rewrites: 17
result Bool: (true).Bool
==========================================
reduce in SMT-VAR-GEN : fresh?(b(0), < "a",1,2 >) .
rewrites: 17
result Bool: (false).Bool
==========================================
reduce in SMT-VAR-GEN : fresh?(b(1), < "a",1,2 >) .
rewrites: 17
result Bool: (false).Bool
==========================================
reduce in SMT-VAR-GEN : fresh?(b(2), < "a",1,2 >) .
rewrites: 17
result Bool: (true).Bool
Bye.
